added authz tests for more test coverage

aryan16 opened the pull request:
**Added new authz tests to improve the test coverage**
AuthZ.rules.to.operation.notPorts
AuthZ.rules.to.operation.notHosts
AuthZ.rules.to.operation.notMethods
Authz.rules.from.source.notRequestPrincipals
AuthZ.rules.from.source.ipBlocks
AuthZ.rules.from.source.notIpBlocks

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#38136)

aryan16 committed:
added authz tests for more test coverage

istio-testing commented:
@aryan16: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio | 27e3a37e15d265b555c41b4a12dd01d6db34bb61 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/38138/lint_istio/1507562574957252608) | true | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

aryan16 committed:
fixed the linter issue

istio-testing commented:
@aryan16: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-security-multicluster_istio | 6cd7ef66b7cd046ad389ae9570e911e0a1321d74 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/38138/integ-security-multicluster_istio/1507565779376148480) | true | `/test integ-security-multicluster_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

aryan16 committed:
multicluster test failure fix

istio-testing commented:
@aryan16: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-security-multicluster_istio | a04054b48589dccac60e7ee5f3985ba95984eed9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/38138/integ-security-multicluster_istio/1507586905628217344) | true | `/test integ-security-multicluster_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#38137)

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#38144)

ramaraochavali committed:
move cache key build outside of critical section (#38149)

* move cache key build outside of critical section

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* remove unused field

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix ut

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

Commit:
add externalIPs in helm chart gateway service (#37904)

istio-testing committed:
Automator: update istio/api@master dependency in istio/istio@master (#38154)

aryan16 committed:
updated authz tests

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#38155)

howardjohn committed:
gateway: expose mechanism to use spiffe workload cert at gateway (#36922)

There are various times when users need to deploy TLS on a workload, but
doesn't care at all what certificate is used. One example could be
istio.io docs, but that is a minor case. The more important is things
like
https://aws.amazon.com/blogs/containers/secure-end-to-end-traffic-on-amazon-eks-using-tls-certificate-in-acm-alb-and-istio/.

Folks will run `client ---TLS---> Cloud LB (terminate) --TLS---> Istio
Ingress (terminate) ---> backend`.

The connection between cloud LB and Istio Ingress doesn't do any
verification, which leads to using self signed certs. While there is
technically nothing wrong with this, it adds extra steps and draws
scrutiny from security-conscious users.

As a step to make this easier, we can allow using the workloads mTLS
certs. This works in `MUTUAL` and `SIMPLE` cases.

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#38156)

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#38157)

howardjohn committed:
pilot: unify and improve sub/unsub in delta and sotw (#37989)

* pilot: unify and improve sub/unsub in delta and sotw

Currently:
* SotW always does a full push on request
* Delta does a partial push only on subscribe, but not on unsubscribe.
* SotW will never skip a push, and will instead send a push with 0
  resources. Delta will skip a push only for EDS

New behavior:
* Both Delta and SotW will do partial pushes for subscribes.
* Both Delta and SotW, If there are only unsubcribes, we will not send a response at all

This optimizes request cases. For example, say we have 100 listeners and
remove 10. 45s later, those listeners will drain. This results in an XDS
requests with 99 routes, 98, 97 ... 90 routes - about 1000 routes
generated in total. With this optimization, we will send 0.

On the other side, say we add 1 listener. Before, we would push RDS
(100), and then immediately get a RDS request and send all 101 again.
With this change, we would send the same 100 in the initial RDS push,
but on the request we would just send the new 1 requested.

* fix sds

* address comments

howardjohn committed:
Revert "The enhancement for function IsIPv6Proxy (#37654)" (#38161)

This reverts commit da38ba9a3041323fdc9658ddb6d9c5f7d2bcb2f8.

Commit:
fix debug panic (#38163)

kfaseela committed:
workloadSelector support in DR (#37174)

* [WIP] workloadSelector support in DR

Signed-off-by: Faseela K <faseela.k@est.tech>

* Update api dependency after rebase of api PR

Signed-off-by: Faseela K <faseela.k@est.tech>

* make destionationRuleIndex accept multiple DRs for a host

Signed-off-by: Faseela K <faseela.k@est.tech>

* Fix rebase errors

Signed-off-by: Faseela K <faseela.k@est.tech>

* Rebase on top of DR refactoring PR

Signed-off-by: Faseela K <faseela.k@est.tech>

* Fix issues with rebase of sidecar.go

Signed-off-by: Faseela K <faseela.k@est.tech>

* add unit tests

Signed-off-by: Faseela K <faseela.k@est.tech>

* add release-notes

Signed-off-by: Faseela K <faseela.k@est.tech>

* Incorporate review comments

Signed-off-by: Faseela K <faseela.k@est.tech>

* Rebase and add map equals function

Signed-off-by: Faseela K <faseela.k@est.tech>

* incorporate review comments and add more unit tests

Signed-off-by: Faseela K <faseela.k@est.tech>

* Add unit tests for workloadselector drs in DestinationRuleInheritance

Signed-off-by: Faseela K <faseela.k@est.tech>

* Incorporate review comments and rebase

Signed-off-by: Faseela K <faseela.k@est.tech>

* Enhance readability of mergeDestinationRule() logic

Signed-off-by: Faseela K <faseela.k@est.tech>

* Remove custom api references - api PR is merged

Signed-off-by: Faseela K <faseela.k@est.tech>

* Rebase

Signed-off-by: Faseela K <faseela.k@est.tech>

* Incorporate review comments from Zhonghu

Signed-off-by: Faseela K <faseela.k@est.tech>

* additional review comments and rebase

Signed-off-by: Faseela K <faseela.k@est.tech>

* Rebase and add additional validation tests

Signed-off-by: Faseela K <faseela.k@est.tech>

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#38168)

ZhengzheYang committed:
add enable install env var to be able to skip installation entirely (#38158)

Change-Id: I9e303287bbdb82c22b40590fef9092e77b7d9df4

hzxuzhonghu committed:
Clean up unneeded method (#38140)

* Clean up un needed method

* rebaseing

ramaraochavali committed:
use same ipv6 detection logic every where (#38100)

* use same ipv6 detection logic every where

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix test

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix ut

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* bc

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

Commit:
workaround lack of GNU date on OSX (#38066)

Change-Id: I752c5d0f1b70a2ed2ce2aeb2609c58e67fac50e7

ramaraochavali committed:
refactor xds code (#38141)

* refactor xds code

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* move delta back

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* revert authenticate

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* revert delta

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

ramaraochavali committed:
Revert "use same ipv6 detection logic every where (#38100)" (#38178)

This reverts commit 8d13c070def1510971b0779007d8901cd4acadec.

vishalpowar committed:
Update gcp mesh-name to 'td.networking.gke.io/mesh-name' (#38185)

* Changes to make pilot-agent generate correct bootstrap to connect to GCP
TrafficDirector

The changes in the PR fall in two categories
1) bootstrap template (gcp_envoy_bootstrap.json)
  - Change to support the new xDS V3 nodeID format used by
  TrafficDirector
  - Change to explictly support V3 API when connecting to
  TrafficDirector
  - Changes to support istio-proxy inbound interception.

2) supporting changes in pilot-agent to set bootstrap template
varaibles.

cla: yes

* Pass the metadata.Labels to the bootstrap template and change
gcp_envoy_bootstrap file to use information to generate correct NodeId

* Pass the metadata instead of just label to the template.

Also, update the gcp_envoy_bootstrap label lookup from "gke.io/mesh" to "td.io/scope"

* Remove the metadata entries for TRAFFICDIRECTOR_ as this will come from
sidecar injector

* Update the gcp_envoy_bootstrap.json to use correct label for looking up
mesh-name

* Update fcp_envoy_bootstrap.json to use the correct label 'networking.gke.io/mesh/mesh-name'

* Update gcp mesh-name to 'td.networking.gke.io/mesh-name'

howardjohn committed:
gateway-api: run conformance tests (#38116)

* gateway-api: run conformance tests

* lint

* fix rebase

* Install CRDs

* fix race

* fix

* feature label

bianpengyuan committed:
Remove meaningless CA authentication check. (#38188)

Commit:
Update openssf badge (#38189)

* Update openssf badge

Shriram set up a CII badge in 2017, so I thought it was about time to update it.

* Update README.md

litong01 committed:
Added warning for setupkind script (#38181)

* Added warning for setupkind script

setupkind script can only successfully setup kind
with metallb in Linux environment. This PR will warn
a user if this script gets run in an envrionment other
than Linux

Signed-off-by: Tong Li <litong01@us.ibm.com>

* setupkind script can only successfully setup kind
with metallb in Linux environment. This PR will warn
a user if this script gets run in an envrionment other
than Linux

Signed-off-by: Tong Li <litong01@us.ibm.com>

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#38203)

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#38205)

Commit:
CA integration through Envoy SDS (#37947)

* CA integration through SDS:
 - Make agent detect socket and certs files on fixed paths.
 - Envoy bootstrap template points to fixed socket path.
 - Unit tests.

Signed-off-by: Max Lambrecht <max.lambrecht@hpe.com>

* CA integration through SDS:
- Add workload-socket and workload-certs volumes and mounts to testdata files.

Signed-off-by: Max Lambrecht <max.lambrecht@hpe.com>

* CA integration through SDS:
- Adding details to Agent README.

Signed-off-by: Max Lambrecht <max.lambrecht@hpe.com>

* CA integration through SDS:
- Add workload-socket and workload-certs volumes and mounts to manifest charts.

Signed-off-by: Max Lambrecht <max.lambrecht@hpe.com>

* CA integration through SDS: Add release note

* Revert changes in testdata

* Applying gofumpt

* Amend markdown

* Amend markdown

* Applying make gen changes

* Applying make gen changes

* Rename certs volume to gke-workload-certificate for GKE

* Add {} to emptyDir volumes

* Removing unnecessary volumes.

* Applying 'make gen' changes

* Rename release notes file

* Add detail in istio-agent/README.md

* Add socket health check. Refactor init SDS Server logic in Agent run method

* Rename release note file

* Add Agent test with Unhealthy SDS socket scenario. Fix checkSocket() method

* Change log line to debug level

nmittler committed:
[TF] Echo: add 2-ns view and fix delta XDS version (#38202)

The 2-namespace echo deployment view will be used by up-coming security tests.

howardjohn committed:
Skip GW conformance for now (#38208)

* Skip GW conformance for now

* done

howardjohn committed:
Fix date command in logging (#38209)

rveerama1 committed:
Handle PrivateKeyProvider configuration (#37681)

* Handle PrivateKeyProvider configuration

When the PrivateKeyProvider configuration information is provided
through Istio operator yaml file, parse and pass it to gateway or
sidecars. Envoy will act based on the information provided by the
configuration.

To set the mesh wide defaults, configure the `defaultConfig` section of `meshConfig`.
For example:

meshConfig:
  defaultConfig:
    privateKeyProvider:
      cryptomb:
        pollDelay: 0.01s

This can also be configured on a per-workload basis by configuring the `proxy.istio.io/config`
annotation on the pod.
For example:

annotations:
  proxy.istio.io/config: |
    privateKeyProvider:
      cryptomb:
        pollDelay: 0.01s

* add release notes

* add validation and tests

* modify according to api change

* address comments

* Fix lint issues

* address comments

* address jacob-delgado and howardjohn comments

* add tests

* use default config

* fix rebase conflicts

* fix config for gateway cert

* fix rebase conflicts

howardjohn committed:
vm tests: update and deprcated OS's tested (#38182)

* vm tests: update and deprcated OS's tested

* fix

* fix docker.xyz

* cleanup

yangminzhu commented on tests/integration/security/testdata/authz/v1beta1-operation.yaml.tmpl:
```diff
@@ -0,0 +1,49 @@
+    - operation:
+        notHosts: ["deny.com"]
+---
+# The following policy won't allow request from port 8091
```
port is the destination, so it's not `from port 809` but `to port 8091`.

yangminzhu commented on tests/integration/security/testdata/authz/v1beta1-operation.yaml.tmpl:
```diff
@@ -0,0 +1,49 @@
+    - operation:
+        notMethods : ["GET"]
+---
+# The following policy won't allow request from hosts "deny.com"
```
hosts is the destination, so it's not `from hosts "deny.com"` but `to host deny.com`

yangminzhu commented on tests/integration/security/authorization_test.go:
```diff
@@ -435,6 +437,81 @@ func TestAuthorization_Deny(t *testing.T) {
 }
 
+// TestAuthorization_NegativeOperation tests the authorization policy with negative match.
+func TestAuthorization_NegativeOperation(t *testing.T) {
```
There is already a `TestAuthorization_NegativeMatch()` for negative matching fields, could you update it instead of adding a new one?

hzxuzhonghu committed:
Fix missing dr dependency when merging (#38088)

nmittler committed:
Restore Set.InsertAll (#38204)

This was removed by https://github.com/istio/istio/pull/38140, but should not have been. The Insert/InsertAll separation avoids a needless array allocation for the vast majority of Insert calls.

howardjohn committed:
Fix VM postsubmit jobs (#38212)

In a recent PR, we fixed the VM tests to actually run the different
images - previously they always ran the same image. This exposed another
bug, that we skipped building these images entirely as well. Start
building them again.

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#38221)

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#38220)

howardjohn committed:
Skip CentOS 8 tests for VMs (#38225)

See https://github.com/istio/istio/issues/38224

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#38223)

bianpengyuan committed:
Implement Wasm pull secret. (#37161)

* Implement Wasm pull secret.

* rn

* ut

* update.

* test.

* tidy

* gen

* update.

* update.

* test.

* Update wasm-secret.yaml

* update.

* fix test.

* lint.

* update

* update.

* integration test.

* update.

* update.

* fix.

* update test.

* gen.

* Add prometheus dump.

* update.

* fix.

* update.

* fix

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#38227)

bianpengyuan committed:
Add comment about Wasm env clean up. (#38228)

ramaraochavali committed:
replace deprecated header match specifier (#38196)

* remove deprecated header match specifier

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* minor change

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* rever header

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* revert security

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

bianpengyuan committed:
Deploy container registry before echo. (#38239)

zhlsunshine committed:
mock net.InterfaceAddrs to make its unit test become available (#38195)

* mock net.InterfaceAddrs to make its unit test become available

* change the unit test in corrected format

* remove the unexception unit test name

lei-tang committed:
TLS Configuration API (Phase 1) (#38186)

* TLS Configuration API (Phase 1)

This PR implements the phase 1 TLS configuration API, including its unit tests and integration tests.

* Refactor based on comments

* Simplified getTLSVersions()

* Use the original getTLSVersions() due to the lint error of the
simplified getTLSVersions()

* Refactor get TLS version function

* Remove integration test

* Refactor getMinTLSVersion()

AdamKorcz committed:
Fix broken OSS-fuzz build (#38243)

Commit:
tf: noisy log (#38251)

Change-Id: I952140b9356cabcdbbd0eac0d6472ee59b6b61b8

hzxuzhonghu committed:
Refactor workloadentry controller  (#38232)

* Reafctor workloadentry controller to use the generic queue insteadd of k8s lib

* update

* update

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#38255)

istio-testing committed:
Automator: update istio/client-go@master dependency in istio/istio@master (#38256)

Commit:
only get secretcontroller when needed (#38257)

Commit:
syncz support ECDS (#38165)

pinxiong committed:
fix: Fix the typo comment (#38263)

howardjohn committed:
Bump iptables image (#38211)

* Bump iptables image

* Update Dockerfile.proxyv2

ramaraochavali committed:
fix incorrect comment (#38260)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

nmittler committed:
[tf] Optimize scope close (#38270)

We currently synchronously run all added closers, which can take a long time. Common use case for a closer is deleting a YAML file that was applied during a test.

This PR runs all closers in parallel. Significantly improves shutdown time on hosted k8s (e.g. GKE).

hzxuzhonghu committed:
Fix root cert configmap err check (#38231)

Commit:
move apiTypePrefix to upstream lib (#38259)

kfaseela committed:
Add auto-sni support (#38238)

* [WIP] Add auto-sni support

Signed-off-by: Faseela K <faseela.k@est.tech>

* Fix unit test failure

Signed-off-by: Faseela K <faseela.k@est.tech>

* enhance release-notes and small code improvements

Signed-off-by: Faseela K <faseela.k@est.tech>

istio-testing committed:
Automator: update proxy@master in istio/istio@master (#38275)

kfaseela committed:
Fix typo in auto-sni-support releasenotes (#38276)

Signed-off-by: Faseela K <faseela.k@est.tech>

zhlsunshine committed:
istioctl miss to append ipv6 domain when using istioctl pc route (#38262)

* istioctl miss to append ipv6 domain when using istioctl pc route

* add test case for describeRouteDomains

Commit:
istioctl ps support ECDS (#38273)

* istioctl ps support ECDS

* add release-notes

Commit:
create-remote-secret: handle k8s 1.24+ sa token behavior (#38249)

* create-remote-secret: handle k8s 1.24+ sa token behavior

Change-Id: Idb56e828b0d0d22e3a03a233f2d809d0d73be894

* fix comment

Change-Id: I2ddb1b3e921627156b6da3df720f9871378cc45c

* fix tests

Change-Id: I4e82c910f7650436f4b019bc47e189e30a794699

Commit:
Add the prefix to td mesh name in gcp_envoy_bootstrap.json (#38264)
